### PR TITLE
support for sensors: SHT3x, Si7021, Si7021 for Sonoff, MAX6675 K

### PR DESCRIPTION
**Pull request by krycha88**

### CODE REVIEW

File: libraries/SuplaDevice/src/supla/sensor/SHT3x.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+#ifndef _sht3x_h
+#define _sht3x_h
+
+// Dependency: Risele SHT3x library - use library manager to install it
+// https://github.com/closedcube/ClosedCube_SHT31D_Arduino
+
+#include "ClosedCube_SHT31D.h"
```
Comment (klew) [inline libraries/SuplaDevice/src/supla/sensor/SHT3x.h]: use < > for files which are not in the same folder as compiled file. Apply the same comment to other files

File: libraries/SuplaDevice/src/supla/sensor/Si7021.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    if (isnan(value)) {
+      value = HUMIDITY_NOT_AVAILABLE;
+    }
+
+    return value;
+  }
+
+  void onInit() {
+    sensor = Adafruit_Si7021();
```
Comment (klew) [inline libraries/SuplaDevice/src/supla/sensor/Si7021.h]: I think this is not needed and not correct operation. In Si7021 example it is called on a line where object is declared, so it is just explicite call to default constructor. In your case there is assignement operator used.

File: libraries/SuplaDevice/src/supla/sensor/Si7021_sonoff.h [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+#include <supla/sensor/therm_hygro_meter.h>
+
+namespace Supla {
+namespace Sensor {
+class Si7021Sonoff : public ThermHygroMeter {
+ public:
+  Si7021Sonoff(int pin) {
+    _pin = pin;
+    pinMode(_pin, INPUT);
+    delay(100);
```
Comment (klew) [inline libraries/SuplaDevice/src/supla/sensor/Si7021_sonoff.h]: Move those operations to "onInit" method

File: libraries/SuplaDevice/src/supla/sensor/Si7021_sonoff.h [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+    return value;
+  }
+
+  void iterateAlways() {
+    if (lastReadTime + 10000 < millis()) {
+      lastReadTime = millis();
+      channel.setNewValue(getTemp(), getHumi());
+    }
+  }
```
Comment (klew) [inline libraries/SuplaDevice/src/supla/sensor/Si7021_sonoff.h]: please don't copy bad examples ;). This method is already implemented properly in base class, so you can remove it from here completely.

File: libraries/SuplaDevice/src/supla/sensor/Si7021_sonoff.h [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    channel.setNewValue(getTemp(), getHumi());
+  }
+
+ private:
+  bool ReadTemp() {
+    _temp = NAN;
+    _humidity = NAN;
+
+    uint8_t d[5];
+    d[0] = d[1] = d[2] = d[3] = d[4] = 0;
```
Comment (klew) [inline libraries/SuplaDevice/src/supla/sensor/Si7021_sonoff.h]: you can init it with nulls via:
uint8_t d[5] = {0};
or
uint8_t d[5] = {};

**Review by klew — CHANGES_REQUESTED** [review]
In case of those sensors, there are common methods to read temperature and humidity from device, so it looks quite strange that we try to read it twice from device (first time in getTime, second in getHumi).
You could override iterateAlways method and put there single "readValuesFromDevice" method which stores temperature and humidity in class variables, then call channel.setNewValue(temperature, humidity); instead of using getTemp and getHumi, which doesn't fit well to those sensors.

**Comment by klew** [comment]
Thanks for changes. I'm ok with accepting this PR.

File: libraries/SuplaDevice/src/supla/sensor/MAX6675_K.cpp [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+  value = spiRead();
+  value <<= 8;
+  value |= spiRead();
+
+  digitalWrite(pin_CS, HIGH);
+
+  if (value & 0x4) {  // this means there is no probe connected to Max6675
+    Serial.print(F("no probe connected to Max6675"));
+    return -111;
```
Comment (klew) [inline libraries/SuplaDevice/src/supla/sensor/MAX6675_K.cpp]: Why -111? Maybe TEMPERATURE_NOT_AVAILABLE will be better here.

**Comment by klew** [comment]
👍

**Comment by ermaya** [comment]
The idea was to show that the connection between Max6675 and the probe is interrupted "open circuit" as a reference to the user.
As the reading range is from 0º to 1014º any negative value could be used for this.

**Pull request merged**